<?xml version="1.0"?>
<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>

  <arg name="model" default="$(find p3dx_description)/urdf/pioneer3dx.xml"/>
  <param name="robot_description" textfile="$(arg model)" />

  <node pkg="stage_ros" type="stageros" name="stageros"
args="$(find robotformations)/launch/demo_obl.world" respawn="false">
    <param name="base_watchdog_timeout" value="0.2"/>
  </node>

 <!--- Run map server and rviz
<node name="map_server" pkg="map_server" type="map_server"
  args="$(find robotformations)/launch/ISRsea1_map.pgm 0.01" respawn="false"/>
<node pkg="rviz" type="rviz" name="rviz" args="-d $(find robotformations)/rviz_configs/stage_multi_p3dx.rviz"/>-->

 <!-- LEADER ROBOT -->
<group ns="robot_0">
  <arg name="tf_prefix" value="robot_0" />
  
  <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop" output="screen"> <!--launch-prefix="xterm -e">-->
  </node>

  <include file="$(find p3dx_description)/launch/p3dx_description.launch">
   <arg name="tf_prefix" value="$(arg tf_prefix)" />
  </include>

  <node pkg="fake_localization" type="fake_localization" name="fake_localization" output="screen">
    <param name="global_frame_id" value="/map"/>
    <param name="odom_frame_id" value="$(arg tf_prefix)/odom"/>
    <param name="base_frame_id" value="$(arg tf_prefix)/base_footprint"/>
  </node>
</group>

<!-- FOLLOWER ROBOTS -->
<group ns="robot_1">
  <arg name="tf_prefix" value="robot_1" />

  <node pkg="robotformations" type="rf_node_switch_column2oblique_simCamera" name="rf_node_switch_column2oblique_simCamera" >
    <param name="instampname" value="/robot_1" />
    <param name="incvrep" value="0.9" />
    <param name="incvatt" value="0.5" />
    <param name="seconds2switch" value="15" />
    <param name="inleaderstampname_obl" value="/robot_0" />
    <param name="inlid_obl" value="1.0" />
    <param name="inlambda_obl" value="1.2" />
    <param name="intau_delta_obl" value="0.3" />
    <param name="inmu_obl" value="10.0" />
    <param name="ingamma_obl" value="PI/4" />
    <param name="inangle_desired_obl" value="0.30" />
    <param name="inleaderstampname_col" value="/robot_0" />
    <param name="inlid_col" value="1.0" />
    <param name="inlambda_col" value="1.2" />
    <param name="intau_delta_col" value="0.3" />
  </node>

  <include file="$(find p3dx_description)/launch/p3dx_description.launch">
   <arg name="tf_prefix" value="$(arg tf_prefix)" />
  </include>

  <node pkg="fake_localization" type="fake_localization" name="fake_localization" output="screen">
    <param name="global_frame_id" value="/map"/>
    <param name="odom_frame_id" value="$(arg tf_prefix)/odom"/>
    <param name="base_frame_id" value="$(arg tf_prefix)/base_footprint"/>
  </node>
</group>

<group ns="robot_2">
  <arg name="tf_prefix" value="robot_2" />
  
  <node pkg="robotformations" type="rf_node_switch_column2oblique_simCamera" name="rf_node_switch_column2oblique_simCamera" >
    <param name="instampname" value="/robot_2" />
    <param name="incvrep" value="0.9" />
    <param name="incvatt" value="0.5" />
    <param name="seconds2switch" value="15" />
    <param name="inleaderstampname_obl" value="/robot_1" />
    <param name="inlid_obl" value="1.0" />
    <param name="inlambda_obl" value="1.2" />
    <param name="intau_delta_obl" value="0.3" />
    <param name="inmu_obl" value="10.0" />
    <param name="ingamma_obl" value="PI/4" />
    <param name="inangle_desired_obl" value="0.30" />
    <param name="inleaderstampname_col" value="/robot_1" />
    <param name="inlid_col" value="1.0" />
    <param name="inlambda_col" value="1.2" />
    <param name="intau_delta_col" value="0.3" />
    <remap from="scan" to="base_scan"/>
  </node>

  <include file="$(find p3dx_description)/launch/p3dx_description.launch">
   <arg name="tf_prefix" value="$(arg tf_prefix)" />
  </include>

  <node pkg="fake_localization" type="fake_localization" name="fake_localization" output="screen">
    <param name="global_frame_id" value="/map"/>
    <param name="odom_frame_id" value="$(arg tf_prefix)/odom"/>
    <param name="base_frame_id" value="$(arg tf_prefix)/base_footprint"/>
  </node>
</group>

</launch>